app/models/concerns/foreman_remote_execution/host_extensions.rb

```diff
@@ -3,17 +3,55 @@ module HostExtensions
     extend ActiveSupport::Concern
 
     included do
+      alias_method_chain :build_required_interfaces, :remote_execution
+      alias_method_chain :reload, :remote_execution
+      alias_method_chain :becomes, :remote_execution
+
       has_many :targeting_hosts, :dependent => :destroy, :foreign_key => 'host_id'
     end
 
-    # create or overwrite instance methods...
-    def instance_method_name
+    def execution_interface
+      get_interface_by_flag(:execution)
     end
 
-    module ClassMethods
-      # create or overwrite class methods...
-      def class_method_name
+    def remote_execution_proxies(provider)
+      proxies = {}
+      proxies[:subnet]   = execution_interface.subnet.remote_execution_proxies.with_features(provider) if execution_interface && execution_interface.subnet
+      proxies[:fallback] = smart_proxies.with_features(provider) if Setting[:remote_execution_fallback_proxy]
+
+      if Setting[:remote_execution_global_proxy]
+        proxy_scope = if Taxonomy.enabled_taxonomies.any?
+                        ::SmartProxy.with_taxonomy_scope_override(organization, location)
```

Comment:
https://github.com/theforeman/foreman/blob/develop/app/models/concerns/taxonomix.rb#L38  - reversed order of taxonomies is expected: `location, organization`